### PR TITLE
Add terraform documentation

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Add terraform documentation

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/docs/Terraform.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Terraform
+
+Terraform is used by sumologic kubernetes collection during setup process
+to automatically create HTTP sources and to store their URLs in the cluster's secret.
```
Comment (sumo-drosiek) [inline deploy/docs/Terraform.md:4]:
```suggestion
to automatically create HTTP sources and to store their URLs in the kubernetes' secret.
```

File: deploy/docs/Terraform.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  sources:
+    # ...
+    logs:
+      # ...
+      default:
+        # ...
+        properties:
+          fields:
+            node: hornetq-livestream-9
+            demployment: sumologic
```
Comment (vsinghal13) [inline deploy/docs/Terraform.md:84]:
```suggestion
            deployment: sumologic
```

File: deploy/docs/Terraform.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    token: "${file(\"/var/run/secrets/kubernetes.io/serviceaccount/token\")}"
+    load_config_file: false
+    # exec:
+    #   api_version:
+    #   command:
+    #   args: []
+    #   env: {}
+```
+
+**Note** [Documentation for kuberenetes provider](https://www.terraform.io/docs/providers/kubernetes/index.html)
```
Comment (samjsong) [inline deploy/docs/Terraform.md:40]:
```suggestion
**Note** [Documentation for Kubernetes provider](https://www.terraform.io/docs/providers/kubernetes/index.html)
```

File: deploy/docs/Terraform.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## Sumo Logic terraform provider
```
Comment (samjsong) [inline deploy/docs/Terraform.md:42]:
```suggestion
## Sumo Logic Terraform provider
```
nit: capitalization for consistency

File: deploy/docs/Terraform.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# Terraform
+
+Terraform is used by sumologic-kubernetes-collection during the setup process
+to automatically create HTTP sources and store their URLs in the Kubernetes secret.
+
+We are using two providers to perform those actions:
+ * [Kubernetes Terraform provider](https://www.terraform.io/docs/providers/kubernetes/)
+ * [Sumo Logic Terraform provider](https://www.terraform.io/docs/providers/sumologic/)
+
+## Kubernetes terraform provider
```
Comment (samjsong) [inline deploy/docs/Terraform.md:10]:
```suggestion
## Kubernetes Terraform provider
```
nit: capitalization for consistency

File: deploy/docs/Terraform.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+[Kubernetes terraform provider](https://www.terraform.io/docs/providers/kubernetes/) is responsible for creating the secret with the created HTTP source endpoints during setup process. The default configuration is expected to work in most cases, however, for self-hosted kubernetes clusters there can be a few exceptions. For these cases we expose the provider configuration in `values.yaml`.
```
Comment (samjsong) [inline deploy/docs/Terraform.md:12]:
```suggestion
[Kubernetes Terraform provider](https://www.terraform.io/docs/providers/kubernetes/) is responsible for creating the secret with the created HTTP source endpoints during setup process. The default configuration is expected to work in most cases, however, for self-hosted kubernetes clusters there can be a few exceptions. For these cases we expose the provider configuration in `values.yaml`.
```
nit: capitalization for consistency

**Pull request merged**